node_modules/oae-authentication/lib/api.js

```diff
@@ -674,6 +683,118 @@ var getUserIdFromLoginId = module.exports.getUserIdFromLoginId = function(tenant
     _getUserIdFromLoginId(loginId, callback);
 };
 
+
+/**
+ * Get the secret for an existing user's resetpassword request
+ *
+ * @param  {Context}    ctx               Standard context object containing the current user and the current tenant
+ * @param  {String}     userName		  The user's own unique username 
+ * @param  {Function}   callback          Standard callback function
+ * @param  {Object}     callback.err      An error that occurred, if any
+ */
+var getResetPasswordSecret =module.exports.getResetPasswordSecret = function(ctx, tenantAlias, username, callback){
+	
+	// associate the userID with loginID for later operations
+    tenantAlias = tenantAlias || ctx.tenant().alias;
+
+    // Create the loginid object first, so it can be passed into the validation
+    var loginId = new LoginId(tenantAlias, AuthenticationConstants.providers.LOCAL, username);
+	
+	// generate a secret token
+	var secret = crypto.randomBytes(16).toString('hex');
+	
+	// check if we can get an existing userID using the userName the user provided
+	_getUserIdFromLoginId(loginId, function(err, userId) {
+		
+        if (userId) {
+	
+			// store it to cassandra database with 24hr valid time USING CONSISTENCY QUORUM AND TTL 86400 
+            Cassandra.runQuery('UPDATE "AuthenticationLoginId" USING TTL 86400 SET "secret" = ? WHERE "loginId" = ?', [secret, _flattenLoginId(loginId)], function(err) {
+		        if (err) {
+		            log().error({'err': err}, 'Error in creating a secret token');
+		            return callback(err);
+		        }
+		        return callback();
+			});
+
+			// get user's object as we will try to send the user an email with token
+			PrincipalsAPI.getUser(ctx, userId, function(err, user) {
```

Comment:
I think you'll have to use the `PrincipalsDAO` here. If the current context is anonymous, the `email` property will be scrubbed from the `user` profile.